pkg/lightning/backend/tidb/tidb.go

```diff
@@ -411,27 +492,50 @@ func (be *tidbBackend) WriteRowsToDB(ctx context.Context, tableName string, colu
 		insertStmt.WriteByte(')')
 	}
 	insertStmt.WriteString(" VALUES")
+	return &insertStmt
+}
 
-	// Note: we are not going to do interpolation (prepared statements) to avoid
-	// complication arise from data length overflow of BIT and BINARY columns
+// Only used for TestWriteRowsErrorSkip
+var mockErrorCount = 0
 
-	for i, row := range rows {
-		if i != 0 {
-			insertStmt.WriteByte(',')
+func (be *tidbBackend) execStmts(ctx context.Context, stmtTasks []stmtTask, rows tidbRows) (int, error) {
+	// Retry will be done externally, so we're not going to retry here.
+	tx, err := be.db.BeginTx(ctx, &sql.TxOptions{})
```

Comment:
Since some if the stmtTask can fail, I think there is no good to exec all the statements in one tx.